### PR TITLE
Handle envelopes stuck in 'created' status

### PR DESCRIPTION
**Pull request by rkondratowicz**
https://tools.hmcts.net/jira/browse/BPS-1065

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessorTest.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+        storageHasBlob(envelope.fileName, envelope.container);
+        dbHas(envelope);
+
+        // when
+        containerProcessor.process(envelope.container);
+
+        // then
+        verifyNoInteractions(blobProcessor);
+    }
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessorTest.java]: Interestingly, unit test for `ContainerProcessor` did not exist until now.
Other tests (not related to continuing processing of envelopes in 'created' status) can be added on a separate PR.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java [hunk 2]
```diff
@@ -71,16 +80,30 @@ private void processBlob(BlobClient blob) {
...
-                        + " ID: {}, filename: {}, container: {}, state: {}",
-                    envelope.id,
-                    envelope.fileName,
-                    envelope.container,
-                    envelope.status.name()
-                ),
+                envelope -> {
+                    if (isStale(envelope)) {
+                        logger.warn(
+                            "Found stale envelope. Resuming its processing. Envelope ID: {}",
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java]: should we also include container and file name?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java]: done. (+ some refactoring)

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java [hunk 3]
```diff
@@ -71,16 +80,20 @@ private void processBlob(BlobClient blob) {
...
+                        blobProcessor.continueProcessing(envelope.id, blob);
+                    } else {
+                        logger.info("Envelope already processed in system, skipping. {}", envelope.getBasicInfo());
+                    }
+                },
                 () -> blobProcessor.process(blob)
             );
+    }
 
+    private boolean isStale(Envelope envelope) {
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java]: Do we need this notion of stale envelope in the context of processing? I though a lease on the blob is what should give you access to the envelope in `CREATED` status.

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java]: good point if you acquire lease we should process it then.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java [hunk 4]
```diff
@@ -71,16 +80,20 @@ private void processBlob(BlobClient blob) {
...
-                    envelope.fileName,
-                    envelope.container,
-                    envelope.status.name()
-                ),
+                envelope -> {
+                    if (isStale(envelope)) {
+                        logger.warn("Found stale envelope. Resuming its processing. {}", envelope.getBasicInfo());
+                        blobProcessor.continueProcessing(envelope.id, blob);
+                    } else {
+                        logger.info("Envelope already processed in system, skipping. {}", envelope.getBasicInfo());
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java]: If the envelope is created but not old enough, this message will be misleading.

What problem does 'stale' solve?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java]: good point.

waiting for 5min to not proceed with envelope that is currently being processed for the first time

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java]: Ok, may save us some unnecessary log entries.

File: src/main/java/uk/gov/hmcts/reform/blobrouter/data/envelopes/Envelope.java [hunk 5]
```diff
@@ -33,4 +33,14 @@ public Envelope(
         this.status = status;
         this.isDeleted = isDeleted;
     }
+
+    public String getBasicInfo() {
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/data/envelopes/Envelope.java]: nice

File: src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java (hunk 3 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java]: fixed

**Comment by rkondratowicz** [comment]
Closing, pipeline issue

**Pull request closed without merging**